whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java

```diff
@@ -184,13 +185,16 @@ private boolean isAllowedIdP(AuthenticationFlowContext context, String clientId,
             try {
                 String apiRealm = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_REALM, context.getRealm().getName());
                 String apiClientId = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_ID, null);
-                String apiClientSecret = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_SECRET, null);
+                String apiClientSecret = getConfigEntry(context, WhitelistAuthenticatorFactory.AUTH_WHITELIST_CLIENT_SECRET, "n/a");
                 List<String> allowedIdPs = client.getListOfAllowedIdPs(clientId, apiRealm, apiClientId, apiClientSecret);
+                logger.tracef("Retrieve allowed IDPs using client %s, realm %s, secret %s from %s", apiClientId, apiRealm, apiClientSecret.substring(0, 3),
+                              client.getUrl());
+                logger.debugf("Retrieved allowed IDPs %s for sp %s from %s", allowedIdPs, providerId);
                 if (allowedIdPs != null && allowedIdPs.contains(providerId)) {
                     isAllowed = true;
                 }
-            } catch (IOException | URISyntaxException e) {
-                logger.errorf(e, "List of assigned IdPs could not fetched clientId=%s, providerId=%s, url=%s", clientId, providerId, client.getUrl());
+            } catch (IOException | URISyntaxException | JSONException e) {
```

Comment:
why catch this RuntimeException?

Reply:
to log it. We had auth issues which did not show up in the logs.